src/Components/CreateImageWizardV2/ImageWizardContext.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useState } from 'react';
+
+import {
+  EnvironmentStateType,
+  filterEnvironment,
+  useGetAllowedTargets,
+} from './steps/ImageOutput/Environment';
+import { GCPAccountTypes } from './steps/TargetEnvironment/GCP/GCPTarget';
+import { useGetAccountData } from './steps/TargetEnvironment/SourcesSelect';
+
+import { RHEL_9, X86_64 } from '../../constants';
+import { ArchitectureItem, Distributions } from '../../store/imageBuilderApi';
+
+type ImageWizardContextPropType = {
+  releaseState: [
+    Distributions,
+    React.Dispatch<React.SetStateAction<Distributions>>
+  ];
+  architectureState: [
+    ArchitectureItem['arch'],
+    React.Dispatch<React.SetStateAction<ArchitectureItem['arch']>>
+  ];
+  environmentState: [
+    EnvironmentStateType,
+    React.Dispatch<React.SetStateAction<EnvironmentStateType>>
+  ];
+  isAwsManualState: [boolean, React.Dispatch<React.SetStateAction<boolean>>];
+  awsSourceState: [
+    [number, string],
+    React.Dispatch<React.SetStateAction<[number, string]>>
+  ];
+  associatedAwsAccountIdState: [
+    string,
+    React.Dispatch<React.SetStateAction<string>>
+  ];
+  isShareGoogleAccountState: [
+    boolean,
+    React.Dispatch<React.SetStateAction<boolean>>
+  ];
+  gcpAccountTypeState: [
+    GCPAccountTypes,
+    React.Dispatch<React.SetStateAction<GCPAccountTypes>>
+  ];
+  gcpAccountEmailState: [string, React.Dispatch<React.SetStateAction<string>>];
+  gcpDomainState: [string, React.Dispatch<React.SetStateAction<string>>];
+  isAzureManualState: [boolean, React.Dispatch<React.SetStateAction<boolean>>];
+  azureSourceState: [
+    [number, string],
+    React.Dispatch<React.SetStateAction<[number, string]>>
+  ];
+  azureTenantIdState: [string, React.Dispatch<React.SetStateAction<string>>];
+  azureSubscriptionIdState: [
+    string,
+    React.Dispatch<React.SetStateAction<string>>
+  ];
+  azureResourceGroupState: [
+    string,
+    React.Dispatch<React.SetStateAction<string>>
+  ];
+};
+
+export const ImageWizardContext =
+  React.createContext<ImageWizardContextPropType>({
```

Comment:
Would turning this into a nested object help organize things a bit? For instance, one level of nesting per wizard step... in other words:
```
{
  firstStep: {},
  secondStep: {},
}
```

Reply:
That's an interesting idea. Have you considered that the fact that we split in steps may not be relevant within the review step ? And when wanting to access a prop to have to search for it?